src/Razor/src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -38,6 +38,13 @@ public static CodeWriter WritePadding(this CodeWriter writer, int offset, Source
                 return writer;
             }
 
+            if (context.SourceDocument.FilePath != null &&
```

Comment:
Where was the actual nullref?

Reply:
Line 82 where it attempts to read absolute indexes based on imports from the source document.